src/navigation/search.py

```diff
@@ -134,6 +148,8 @@ def on_loop(self, context) -> State:
         )
         context.rover.send_drive_command(cmd_vel)
 
-        if context.env.current_fid_pos() is not None and context.course.look_for_post():
-            return approach_post.ApproachPostState()
+        # returns either ApproachPostState, LongRangeState, ApproachObjectState, or None
+        if context.course.check_approach() is not None:
```

Comment:
for safety purposes we should probably save the result in a variable and then just return the variable (in case the check_approach function returns different results across calls)

Reply:
Yes, I first had it saved to a variable, but for some reason it caused some weird behavior of the rover stalling for a little bit before transitioning into one of the states returned by the function. But doing it the way I have here fixed that problem. Maybe there is another underlying issue then.